src/router/BaseRouter.php

```diff
@@ -51,7 +57,9 @@ abstract class BaseRouter<+TResponder> {
   ): (TResponder, ImmMap<string, string>) {
     $method = HttpMethod::coerce($request->getMethod());
     if ($method === null) {
-      throw new MethodNotAllowedException();
+      throw new MethodNotAllowedException(
+        keyset<HttpMethod>(HttpMethod::getValues())
```

Comment:
Is this correct? Do we need to determine which methods are available here (using something like the loop above) instead of returning all of them?

Reply:
this is correct, the request method is not supported by hack-router, therefore we say "X method is not allowed, try Y, Z, W ... etc"

Reply:
Yeah but shouldn't we figure out which methods are supported and only return those? We do that above -- could just extract the code into something like `getAllowedMethods($path)` and reuse it here.